clang/include/clang/Basic/TargetInfo.h

```diff
@@ -1211,6 +1211,19 @@ class TargetInfo : public TransferrableTargetInfo,
       TiedOperand = N;
       // Don't copy Name or constraint string.
     }
+
+    // Output operand bounds can be set by target.
```

Comment:
Maybe make the comment a bit more specific, e.g.
```
// For output operand constraints, the target can set bounds to indicate
// that the result value is guaranteed to fall within a certain range.
// This will cause corresponding assertions to be emitted that will allow
// for potential optimization based of that guarantee.
//
// NOTE: This re-uses the `ImmRange` fields to store the range, which are
// otherwise unused for constraint types used for output operands.
```